### DISCUSSION TITLE
Dynamic import issue when using reverse proxy

### DISCUSSION THREAD

**Discussion by ex-expat** [open-webui/open-webui#6573] [Q&A]
I have an open-webui Docker container deployed on a server and am trying use NGINX as a reverse proxy, but I am having a problem with open-webui's Javascript dynamic imports. For example, if I am proxying from the NGINX container at `localhost:8001` to the open-webui container at `localhost:3000` I get error messages similar to the following:

```
Uncaught (in promise) TypeError: Failed to fetch dynamically imported module: http://localhost:8001/_app/immutable/entry/start.eCizU3Fd.js
```

Open-webui is using the original port since the request is being forwarded via NGINX. Has anyone experienced a similar issue or know how to resolve? I have tried a ton of different proxy settings, etc., and nothing is working.

**Comment by gamesgao** [comment]
Seems I met the same issue here and I'm using Synology Reverse Proxy as a way which I believe that it is also using NGINX.

It used to work in 0.3.32 but after I upgrade to 0.4.3 it stops working. 

Also trying to seek a solution here on what is wrong for the Reserve Proxy Config

**Comment by littlebearz** [comment]
I noticed this issue on v0.5.10, it have to do with dynamic loading the js file in /_app/immutable/*

The fix was for .js files to proxy_pass with the $uri.

upstream @backend {
            server BACKEND_REAL_SERVER_IP_OR_FQDN:8080;
    }

location ~* \.(ico|jpg|jpeg|png|gif|svg|js|css|swf|eot|ttf|otf|woff|woff2|json)$ {
        proxy_pass http://@backend/$uri;
        add_header X-Frame-Options "SAMEORIGIN";
        access_log off;
    }

**Comment by xellos866** [comment]
We have experienced a similar error and maybe our findings will also help you:

After debugging, we have probably found a way to solve the error. It is likely a problem with the number of simultaneous connections from the reverse proxy (nginx) that are sometimes triggered when the login window is opened by the client.

In some cases, up to 150 parallel HTTP connections were being established, which exceeded the limits for parallel connections in our nginx reverse proxy (default was 100). As a result, some of the connections were cut off by nginx. This in turn caused the application not to initialize correctly and triggered the error message.

When reloading, only some of the connections are re-established, and it also seems that there is some short-term caching in Open WebUI itself, so that significantly fewer JS files are sometimes requested in parallel. Furthermore, the error only appeared on fast internet connections, meaning that when we throttled our own connection speed, the error reliably did not occur anymore.

We have now significantly increased the number of allowed parallel connections, and in our tests, we could no longer reproduce the error. 

```
    nginx: {
        worker_connections: 200,
        extra_configs: {
            limit_conn: 'default 200'
        },
```
